directory ~/apptainer/cmd/apptainer outside main module or its selected dependencies

As I try to run command make as the toturial said, I meet this error, how can I solve it?

I'll take up the torch on this, as I'm having this same issue on our HPC.

I've followed the official install instructions posted above, and I receive the following error:

```
[swmgr@toolbox apptainer]$ ./mconfig -p /packages/apptainer/1.2.4/
Configuring for project `apptainer' with languages: C, Golang
=> running pre-basechecks project specific checks ...
=> running base system checks ...
 checking: host C compiler... cc
 checking: host C++ compiler... c++
 checking: host Go compiler (at least version 1.19)... /packages/golang/1.21.3/go/bin/go
 checking: host C compiler option -Wall... yes
 checking: host C compiler option -Werror... yes
 checking: host C compiler option -Wfatal-errors... yes
 checking: host C compiler option -Wno-unknown-warning-option... yes
 checking: host C compiler option -Wstrict-prototypes... yes
 checking: host C compiler option -Wpointer-arith... yes
 checking: host C compiler option -Wbad-function-cast... yes
 checking: host C compiler option -Woverlength-strings... yes
 checking: host C compiler option -Wframe-larger-than=2047... yes
 checking: host C compiler option -Wno-sign-compare... yes
 checking: host C compiler option -Wclobbered... yes
 checking: host C compiler option -Wempty-body... yes
 checking: host C compiler option -Wmissing-parameter-type... yes
 checking: host C compiler option -Wtype-limits... yes
 checking: host C compiler option -Wunused-parameter... yes
 checking: host C compiler option -Wunused-but-set-parameter... yes
 checking: host C compiler option -Wno-discarded-qualifiers... yes
 checking: host C compiler option -Wno-incompatible-pointer-types... yes
 checking: host C compiler option -pipe... yes
 checking: host C compiler option -fmessage-length=0... yes
 checking: host C compiler option -fPIC... yes
 checking: host `ar' path... ar
 checking: host `ld' path... ld
 checking: host `ranlib' path... ranlib
 checking: host `objcopy' path... objcopy
 checking: target C compiler... cc
 checking: target C++ compiler... c++
 checking: target `ar' path... ar
 checking: target `ld' path... ld
 checking: target `ranlib' path... ranlib
 checking: target `objcopy' path... objcopy
 checking: host compiles static binaries... no
 checking: target compiles static binaries... no
 checking: host os type... unix
 checking: host architecture... x86_64
 checking: target architecture... x86_64
 checking: host architecture word size... 64
 checking: target architecture word size... 64
 checking: project version... 1.2.4
 checking: project short version... 1.2.4
 checking: unprivileged user namespaces... enabled. Implying --without-suid
=> running post-basechecks project specific checks ...
 checking: namespace: CLONE_NEWPID... yes
 checking: namespace: CLONE_FS... yes
 checking: namespace: CLONE_NEWNS... yes
 checking: namespace: CLONE_NEWUSER... yes
 checking: namespace: CLONE_NEWIPC... yes
 checking: namespace: CLONE_NEWNET... yes
 checking: namespace: CLONE_NEWUTS... yes
 checking: namespace: CLONE_NEWCGROUP... yes
 checking: feature: NO_NEW_PRIVS... yes
 checking: feature: MS_SLAVE... yes
 checking: feature: MS_REC... yes
 checking: feature: MS_PRIVATE... yes
 checking: user capabilities... yes
 checking: header linux/securebits.h... yes
 checking: header linux/capability.h... yes
 checking: libseccomp+headers... yes
=> generating fragments ...
=> building Makefile ...
=> generating apptainer.spec ...
=> project apptainer setup with :
    - host arch: x86_64
    - host wordsize: 64-bit
    - host C compiler: cc
    - host Go compiler: /packages/golang/1.21.3/go/bin/go
    - host system: unix
      ---
    - target arch: x86_64
    - target wordsize: 64-bit
    - target C compiler: cc
      ---
    - config profile: release
      ---
    - SUID install: no
    - Network plugins: yes
      ---
    - verbose: no
      ---
    - version: 1.2.4
=> /gpfs/packages/apptainer/1.2.4/src/apptainer/builddir/Makefile ready, try:
   $ cd /gpfs/packages/apptainer/1.2.4/src/apptainer/builddir
   $ make
[swmgr@toolbox apptainer]$ cd builddir/
[swmgr@toolbox builddir]$ make
 GEN GO DEP /gpfs/packages/apptainer/1.2.4/src/apptainer/builddir/starter.deps
 GEN GO DEP /gpfs/packages/apptainer/1.2.4/src/apptainer/builddir/apptainer.deps
 GEN /gpfs/packages/apptainer/1.2.4/src/apptainer/scripts/go-generate
 GO apptainer
    [+] GO_TAGS "containers_image_openpgp sylog oci_engine apptainer_engine fakeroot_engine apparmor selinux seccomp"
directory /gpfs/packages/apptainer/1.2.4/src/apptainer/cmd/apptainer outside main module or its selected dependencies
make: *** [Makefile:181: apptainer] Error 1
```

As per the config output above, I'm using golang 1.21.3. This is running on RHEL 8.7:

```
[swmgr@toolbox builddir]$ cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.7 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.7"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.7 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.7
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.7"
```

I've installed all the dependencies:

```
[root@toolbox ~]# yum install -y     libseccomp-devel     squashfs-tools     squashfuse     fuse-overlayfs     fakeroot     /usr/*bin/fuse2fs     cryptsetup     wget git
Updating Subscription Management repositories.
EPEL 8                                                                                          73  B/s | 2.3 kB     00:31
Package libseccomp-devel-2.5.2-1.el8.x86_64 is already installed.
Package squashfs-tools-4.3-20.el8.x86_64 is already installed.
Package squashfuse-0.1.104-1.el8.x86_64 is already installed.
Package fuse-overlayfs-1.9-1.module+el8.7.0+17824+66a0202b.x86_64 is already installed.
Package fakeroot-1.31-1.el8.x86_64 is already installed.
Package e2fsprogs-1.45.6-5.el8.x86_64 is already installed.
Package cryptsetup-2.3.7-2.el8.x86_64 is already installed.
Package wget-1.19.5-10.el8.x86_64 is already installed.
Package git-2.31.1-3.el8_7.x86_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
```

Is there any other output I can provide to help diagnose?

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -146,7 +146,7 @@ You can configure, build, and install Apptainer using the following commands:
 
 ```sh
 ./mconfig
-cd ./builddir
+cd $(/bin/pwd)/builddir
 make
 sudo make install
 ```

